vlib/x/templating/dtm/README.md

```diff
@@ -0,0 +1,278 @@
+# dtm - Dynamic Template Manager
+
+A simple template manager integrated into the V project, designed to combine the power of V
+templates with Vweb, without the need to recompile the application with every change.
+
+## Quick Start
+
+Using the dynamic template manager ( named '**dtm**' in this readme) is relatively straightforward.
+You just need to initialize an instance. Then, call the '**expand**' function in the code that
+manages your template. 
+
+Before starting, You can specify a folder, but it is not mandatory to store the generated cache. If
+nothing is specified or if there is a problem with the targeted folder (for example, a permission
+issue), the DTM will attempt to create a cache folder in the temporary file area of your OS.
+Finally, if all this does not prove successful, then the DTM will disable the cache system and
+notify the user if the cache system was previously required.
+
+However, at the root directory of your project, you need to create a '**templates**' folder. The
+absence of these directories will prevent you from using the dtm. You must add your template files
+into the folder you previously created. ***Be aware that if the templates are not placed in the
+correct directory, the DTM will return an error message indicating that it cannot find the
+template!***
+
+The DTM currently only works with 2 types of files:
+
+- html
+- raw text
+
+Below you will find 2 usage contexts:
+
+### 1. Minimal static generator example :
+
+```v ignore
+import x.templating.dtm
+
+fn main() {
+
+  mut dtmi := dtm.initialize()
+
+    // No need to add this 'defer' if you have chosen to disable the cache system in the options.
+	defer {
+		dtmi.stop_cache_handler()
+	}
+	mut tmp_var := map[string]dtm.DtmMultiTypeMap{}
+	tmp_var['title'] = "V dtm best title" 
+	tmp_var['non_string_type'] = 7
+	tmp_var['html_section_#includehtml'] = '<span>will <br> be <br> escaped <br> in <br> text mode</span>'
+
+	render := dtmi.expand('test.txt', placeholders: &tmp_var)
+
+	println(render)
+
+}
+```
+#### and its template text format :
+
+```
+Title of text : @title
+Value in the text : @non_string_type
+HTML tags is always escaped in text file : @html_section
+```
+
+### 2. Minimal Vweb example:
+
+```v ignore
```

Comment:
Why `v ignore`?

A full example should compile, and it should be as presentable as possible, including formatting etc. It should be ready to be copy/pasted.

Using just v, ensures that `v check-md .` will check that all such examples compile, and are v fmt-ed.